net/src/sim2h_worker.rs

```diff
@@ -0,0 +1,407 @@
+};
+use holochain_tracing::Span;
+use lib3h_zombie_actor::{GhostParentWrapper, GhostCallbackData, WorkWasDone};
+use lib3h_zombie_actor::GhostCanTrack;
+use failure::err_msg;
+use lib3h_protocol::protocol::*;
+use lib3h_protocol::data_types::{GenericResultData, Opaque, SpaceData, StoreEntryAspectData};
+use lib3h_protocol::uri::Lib3hUri;
+use std::convert::TryFrom;
+use detach::Detach;
+use sim2h::crypto::{Provenance, SignedWireMessage};
+use holochain_conductor_api_api::{CryptoMethod, ConductorApi};
+use holochain_json_api::json::RawString;
+
+#[derive(Deserialize, Serialize, Clone, Debug, DefaultJson, PartialEq)]
+pub struct Sim2hConfig {
+    pub sim2h_url: String,
+}
+
+/// removed lifetime parameter because compiler says ghost engine needs lifetime that could live statically
+#[allow(non_snake_case, dead_code)]
+pub struct Sim2hWorker {
+    handler: NetHandler,
+    transport: Detach<TransportActorParentWrapper<Sim2hWorker, GhostTransportWebsocket>>,
+    inbox: Vec<Lib3hClientProtocol>,
+    to_core: Vec<Lib3hServerProtocol>,
+    num_ticks: u32,
+    server_url: Lib3hUri,
+    space_data: Option<SpaceData>,
+    agent_id: Address,
+    conductor_api: ConductorApi,
+}
+
+fn wire_message_into_escaped_string(message: &WireMessage) -> String {
+    let payload: String = message.clone().into();
+    let json_string: JsonString = RawString::from(payload).into();
+    let mut string: String = json_string.into();
+    string = String::from(string.trim_start_matches("\""));
+    string = String::from( string.trim_end_matches("\""));
+    string
+}
+
+impl Sim2hWorker {
+    pub fn advertise(self) -> url::Url {
+        Url::parse("ws://example.com").unwrap()
+    }
+
+    /// Create a new worker connected to the sim2h instance
+    pub fn new(
+        handler: NetHandler,
+        config: Sim2hConfig,
+        agent_id: Address,
+        conductor_api: ConductorApi,
+    ) -> NetResult<Self> {
+        let transport_raw = GhostTransportWebsocket::new(
+            // not used currently inside GhostTransportWebsocket:
+            Address::from("sim2h-worker-transport"),
+            TlsConfig::SuppliedCertificate(TlsCertificate::build_from_entropy()),
+            // not used currently inside GhostTransportWebsocket:\
+            Address::from("sim2h-network"),
+        );
+
+        let mut transport: TransportActorParentWrapper<Sim2hWorker, GhostTransportWebsocket> =
+            GhostParentWrapper::new(
+                transport_raw,
+                "t1_requests", // prefix for request ids in the tracker
+            );
+
+
+        // bind to some port:
+        // channel for making an async call sync
+        debug!("Trying to bind to network...");
+        let (tx, rx) = crossbeam_channel::unbounded();
+        transport.request(
+            Span::todo("Find out how to use spans the right way"),
+            RequestToChild::Bind {
+                spec: Url::parse("wss://localhost:0").expect("can parse url").into(),
```

Comment:
Didn't we decide we need `wss://127.0.0.1:0` to ensure we use ipv4 for now?

Reply:
Hm, good point. Interesting. It seems this was only a problem with the server. This code worked on @zippy's machine with IPv6 interfaces... will change it anyway to be sure.